nbconvert/templates/asciidoc.tpl

```diff
@@ -9,7 +9,14 @@
     +*In[]:*+
 {%- endif -%}
 {%- endif -%}
-[source{% if nb.metadata.language_info %}, {{ nb.metadata.language_info.name }}{% endif %}]
+[source
+{%- if 'magics_language' in cell.metadata  -%}
+   , {{ cell.metadata.magics_language}}
+{%- elif 'pygments_lexer' in nb.metadata.get('language_info', {}) -%}
+   , {{ nb.metadata.language_info.pygments_lexer }}
```

Comment:
Is asciidoc usually rendered by something using pygments for highlighting? We do this for rst because we know that the main processor of rst, docutils, uses pygments. But if it's likely to be processed by another tool, we shouldn't give special status to the pygments highlighter name.